### ISSUE TITLE
Function as attribute

### ISSUE THREAD

**Issue by paulpatarinski** [preactjs/preact-custom-element#14]
Is there an easy way to pass in a function as an attribute? 

Example : 

```
 render({ ondataloaded }) {
      return (
          <button onClick={ondataloaded}>Trigger Data Loaded</button>
      );

registerCustomElement(Clock, "clock", ["ondataloaded"]);

```

**Comment by lucasghizoni** [comment]
You could dispatch an event to outside, like that: 
`this.base.parentElement.dispatchEvent('my-event', new Event())`

, and then, execute your function outside the component, adding an event listener to your custom element reference: 

```
myComponent.addEventListener('my-event', e => {
     // your code 
})
```

**Comment by marvinhagemeister** [comment]
The fix is released as part of `4.0.0` 🎉
